### PR TITLE
use bootstrap.js/app.css again

### PR DESCRIPTION
**Pull request by samthor**
We're seeing odd caching problems in the Real World(™) whereby older HTML files are being served, which request hashed CSS/JS files that don't exist. This was mitigated by #3392 but this PR cleans up the behavior further.

- instead of generating "app-hashed.css" and serving it at all "app.css", "app-whatever.css", this PR now generates _only_ "app.css" and also catches "app-whatever.css".
- we include a cache-busting query param in our layout (e.g., "app.css?v=1234")
- this same exact URL, including cache-busting, is cached by our SW (avoiding #3363)

So if we see odd caching issues, "app.css" and "bootstrap.js" will _still be served_, because the server is ignoring the query string.

This will effect the "bigger refactor" in #3362 but that's fine.

### CODE REVIEW

**Comment by robdodson** [comment]
looks like the test failed trying to do a prod build

**Comment by samthor** [comment]
Tests fixed. PTAL. This can wait until after WDL, because the functionality is the same, it's just a bit less fragile (we don't need to look up _what_ "app-hashedname.css" we currently have).

**Comment by robdodson** [comment]
> We're seeing odd caching problems in the Real World(™) whereby older HTML files are being served, which request hashed CSS/JS files that don't exist.

Before moving forward maybe we should solve this problem. Looking at the MDN docs on cache-control, it looks like setting something to public;max-age=0 (which is what express.static() gives us today) will not prevent intermediaries from caching it. Instead we should set our html to `no-store` https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Preventing_caching

My concern is if we make it so query params always fallback to the current CSS then we could end up with a different kind of mismatching (cache'd HTML is trying to reference some CSS selector that we deleted in the newest CSS).

**Comment by samthor** [comment]
Right but we don't deploy the old CSS. That scenario you posit will happen anyway since those browsers don't have access to the older CSS.

I'm not worried about a user seeing a more up-to-date version of our CSS or JS, even if they have slightly older HTML, although we should try to avoid that too.

File: src/site/_filters/strip-query-params-dev.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+const isProd = process.env.ELEVENTY_ENV === 'prod';
+const noop = (url) => url;
+const strip = (url) => url.split('?')[0];
+
+// Strip query params from URLs in dev only. Used for lazy cache-busting
+// e.g. in non-prod, app.css?blah => app.css
+module.exports = isProd ? noop : strip;
```
Comment (MichaelSolati) [inline src/site/_filters/strip-query-params-dev.js:23]: Ok, so not gonna hold this PR up, but simple question... Could this be a generic filter (not dev specific), where it takes a second argument which would tell the filter if it should strip the params?

```njk
<link rel="stylesheet" href="{{ resourceCSS.path | stripQueryParams(ifDev) }}" />
```

and 

```JS
module.exports =(url, condition) => condition ? noop(url) : strip(url);
```

Comment (samthor) [reply src/site/_filters/strip-query-params-dev.js:23]: Yup. I don't think we'd ever set the boolean to anything but `isDev` though. Happy to change this if we ever need to.

**Comment by robdodson** [comment]
> Right but we don't deploy the old CSS. That scenario you posit will happen anyway since those browsers don't have access to the older CSS.

I'm confused. Why would they request old CSS if we tell the intermediaries to never cache stale html? Wouldn't the html always be current then?

> I'm not worried about a user seeing a more up-to-date version of our CSS or JS, even if they have slightly older HTML, although we should try to avoid that too.

It seems like this could also cause breakages (we remove something in the new CSS or new JS is trying to reference a thing that doesn't exist on the page). Maybe #3362 will resolve it

**Comment by samthor** [comment]
We saw stale HTML in practice.

I'm happy to revisit the partial hashes stuff next week, we can chat about it before merging anything.

**Comment by robdodson** [comment]
> We saw stale HTML in practice.

oh yeah, I think that may be because we weren't using `no-store`. I'm curious if we used `no-store` if we would hit that problem again

**Pull request merged**